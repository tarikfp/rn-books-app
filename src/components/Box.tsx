import { createBox } from '@shopify/restyle';
import * as React from 'react';
import { AppTheme } from '~theme/core';

const ThemedBox = createBox<AppTheme>();
type BoxProps = React.ComponentProps<typeof ThemedBox>;
export const Box: React.FC<BoxProps> = (props) => {
  return <ThemedBox {...props} />;
};
