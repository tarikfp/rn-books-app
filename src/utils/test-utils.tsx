import { ThemeProvider } from '@shopify/restyle';
import { render, RenderOptions } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { theme } from '~theme/core';
import { ThemePreferenceContext } from '~theme/useThemePreference';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ThemePreferenceContext.Provider
      value={{
        isDarkTheme: false,
        toggleTheme: () => {},
      }}
    >
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </ThemePreferenceContext.Provider>
  );
};

const withAllTheProviders = (ui: React.ReactNode) => (
  <AllTheProviders children={ui} />
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const customRenderer = (ui: React.ReactElement) =>
  renderer.create(withAllTheProviders(ui));

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { customRender as render };
// override renderer method
export { customRenderer as createRenderer };
