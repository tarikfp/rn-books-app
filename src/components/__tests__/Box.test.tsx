import React from 'react';
import { Box } from '~components/Box';
import { render } from '~utils/test-utils';

describe('Box Component', () => {
  it('matches snapshot', () => {
    const { toJSON } = render(<Box />);
    expect(toJSON()).toMatchSnapshot();
  });
});
