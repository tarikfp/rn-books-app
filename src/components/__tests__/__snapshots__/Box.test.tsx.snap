// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Component matches snapshot 1`] = `
<View
  style={
    [
      {},
    ]
  }
/>
`;
